app/src/main/java/co/nimblehq/ui/second/SecondViewModel.kt

```diff
@@ -1,37 +1,28 @@
 package co.nimblehq.ui.second
 
 import androidx.hilt.lifecycle.ViewModelInject
-import co.nimblehq.data.lib.schedulers.SchedulersProvider
 import co.nimblehq.ui.base.BaseViewModel
-import co.nimblehq.ui.main.Const
 import co.nimblehq.ui.main.data.Data
 import io.reactivex.Observable
 import io.reactivex.subjects.BehaviorSubject
 
-class SecondViewModel
-@ViewModelInject constructor(
-    private val schedulers: SchedulersProvider
-) : BaseViewModel(), Inputs, Outputs {
+class SecondViewModel @ViewModelInject constructor() : BaseViewModel(), Inputs, Outputs {
 
     private val persistData = BehaviorSubject.create<Data>()
 
     val inputs: Inputs = this
     val outputs: Outputs = this
 
-    init {
-        val dataFromIntent = intent()
-            .subscribeOn(schedulers.io())
-            .map { it.getParcelableExtra<Data>(Const.EXTRAS_DATA) }
-
-        dataFromIntent
-            .subscribe(persistData::onNext)
-            .bindForDisposable()
+    override fun dataFromIntent(data: Data) {
+        persistData.onNext(data)
     }
 
-    override fun setPersistedData() = this.persistData!!
+    override fun setPersistedData() = this.persistData
```

Comment:
Hmmm, a bit of confusing name, no? 🙈  We don't `set` here, but we do `get`

Reply:
Perhaps I'm wrong, I might have been away from Rx too long 😅

Reply:
But weird, why `set` function is in `Outputs`?  Should it be in `Inputs` instead if we really want to set it?